I receive this error when I added <input type="hidden" th:field="*{ID}"/> to create_Npc after doing --> localhost/Npc/createform

There was an unexpected error (type=Internal Server Error, status=500).
_An error happened during template parsing (template: "class path resource [templates/create_Npc.html]")
org.thymeleaf.exceptions.TemplateInputException: An error happened during template parsing (template: "class path resource [templates/create_Npc.html]")_
	at org.thymeleaf.templateparser.markup.AbstractMarkupTemplateParser.parse(AbstractMarkupTemplateParser.java:241)
	at org.thymeleaf.templateparser.markup.AbstractMarkupTemplateParser.parseStandalone(AbstractMarkupTemplateParser.java:100)
	at org.thymeleaf.engine.TemplateManager.parseAndProcess(TemplateManager.java:666)
	at org.thymeleaf.TemplateEngine.process(TemplateEngine.java:1098)
	at org.thymeleaf.TemplateEngine.process(TemplateEngine.java:1072)
	at org.thymeleaf.spring5.view.ThymeleafView.renderFragment(ThymeleafView.java:366)
	at org.thymeleaf.spring5.view.ThymeleafView.render(ThymeleafView.java:190)
	at org.springframework.web.servlet.DispatcherServlet.render(DispatcherServlet.java:1397)
	at org.springframework.web.servlet.DispatcherServlet.processDispatchResult(DispatcherServlet.java:1142)
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1081)
	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:963)
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1006)
	at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:898)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:655)
	at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:883)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:764)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:227)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
	at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
	at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:197)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:97)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:542)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:135)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:78)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:357)
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:382)
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65)
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:893)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1726)
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
	at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191)
	at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: org.attoparser.ParseException: Error during execution of processor 'org.thymeleaf.spring5.processor.SpringInputGeneralFieldTagProcessor' (template: "create_Npc" - line 12, col 46)
	at org.attoparser.MarkupParser.parseDocument(MarkupParser.java:393)
	at org.attoparser.MarkupParser.parse(MarkupParser.java:257)
	at org.thymeleaf.templateparser.markup.AbstractMarkupTemplateParser.parse(AbstractMarkupTemplateParser.java:230)
	... 48 more
Caused by: org.thymeleaf.exceptions.TemplateProcessingException: Error during execution of processor 'org.thymeleaf.spring5.processor.SpringInputGeneralFieldTagProcessor' (template: "create_Npc" - line 12, col 46)
	at org.thymeleaf.processor.element.AbstractAttributeTagProcessor.doProcess(AbstractAttributeTagProcessor.java:117)
	at org.thymeleaf.processor.element.AbstractElementTagProcessor.process(AbstractElementTagProcessor.java:95)
	at org.thymeleaf.util.ProcessorConfigurationUtils$ElementTagProcessorWrapper.process(ProcessorConfigurationUtils.java:633)
	at org.thymeleaf.engine.ProcessorTemplateHandler.handleStandaloneElement(ProcessorTemplateHandler.java:918)
	at org.thymeleaf.engine.TemplateHandlerAdapterMarkupHandler.handleStandaloneElementEnd(TemplateHandlerAdapterMarkupHandler.java:260)
	at org.thymeleaf.templateparser.markup.InlinedOutputExpressionMarkupHandler$InlineMarkupAdapterPreProcessorHandler.handleStandaloneElementEnd(InlinedOutputExpressionMarkupHandler.java:256)
	at org.thymeleaf.standard.inline.OutputExpressionInlinePreProcessorHandler.handleStandaloneElementEnd(OutputExpressionInlinePreProcessorHandler.java:169)
	at org.thymeleaf.templateparser.markup.InlinedOutputExpressionMarkupHandler.handleStandaloneElementEnd(InlinedOutputExpressionMarkupHandler.java:104)
	at org.attoparser.HtmlElement.handleStandaloneElementEnd(HtmlElement.java:79)
	at org.attoparser.HtmlMarkupHandler.handleStandaloneElementEnd(HtmlMarkupHandler.java:241)
	at org.attoparser.MarkupEventProcessorHandler.handleStandaloneElementEnd(MarkupEventProcessorHandler.java:327)
	at org.attoparser.ParsingElementMarkupUtil.parseStandaloneElement(ParsingElementMarkupUtil.java:96)
	at org.attoparser.MarkupParser.parseBuffer(MarkupParser.java:706)
	at org.attoparser.MarkupParser.parseDocument(MarkupParser.java:301)
	... 50 more
Caused by: org.springframework.beans.NotReadablePropertyException: Invalid property 'ID' of bean class [com.senecafoundation.NPCTypes.NPC]: Bean property 'ID' is not readable or has an invalid getter method: Does the return type of the getter match the parameter type of the setter?
	at org.springframework.beans.AbstractNestablePropertyAccessor.getPropertyValue(AbstractNestablePropertyAccessor.java:627)
	at org.springframework.beans.AbstractNestablePropertyAccessor.getPropertyValue(AbstractNestablePropertyAccessor.java:617)
	at org.springframework.validation.AbstractPropertyBindingResult.getActualFieldValue(AbstractPropertyBindingResult.java:104)
	at org.springframework.validation.AbstractBindingResult.getFieldValue(AbstractBindingResult.java:228)
	at org.springframework.web.servlet.support.BindStatus.<init>(BindStatus.java:129)
	at org.springframework.web.servlet.support.RequestContext.getBindStatus(RequestContext.java:903)
	at org.thymeleaf.spring5.context.webmvc.SpringWebMvcThymeleafRequestContext.getBindStatus(SpringWebMvcThymeleafRequestContext.java:227)
	at org.thymeleaf.spring5.util.FieldUtils.getBindStatusFromParsedExpression(FieldUtils.java:306)
	at org.thymeleaf.spring5.util.FieldUtils.getBindStatus(FieldUtils.java:253)
	at org.thymeleaf.spring5.util.FieldUtils.getBindStatus(FieldUtils.java:227)
	at org.thymeleaf.spring5.processor.AbstractSpringFieldTagProcessor.doProcess(AbstractSpringFieldTagProcessor.java:174)
	at org.thymeleaf.processor.element.AbstractAttributeTagProcessor.doProcess(AbstractAttributeTagProcessor.java:74)
	... 63 more

--- a/src/main/java/com/senecafoundation/ChoiceController.java
+++ b/src/main/java/com/senecafoundation/ChoiceController.java
@@ -53,12 +53,13 @@ public class ChoiceController
         model.addAttribute("choice", choice);
         return "create_choice"; 
     }
-    @RequestMapping(value="/updateform", method = RequestMethod.PUT)
-    public String change(@ModelAttribute("choice") Choice choice, BindingResult result, ModelMap model) {
+    @RequestMapping(value="/updateform", method = RequestMethod.POST)
+    public String change(Choice choice, BindingResult result, ModelMap model) {
         if (result.hasErrors()) {
-            return "Error";
+            return "error";
         }
         dataHandler.Update(choice);
+        model.addAttribute("choice",choice);
         return "choice";   
     }
 

--- a/src/main/java/com/senecafoundation/NPCController.java
+++ b/src/main/java/com/senecafoundation/NPCController.java
@@ -20,7 +20,7 @@ import org.springframework.web.bind.annotation.RequestMethod;
 @RequestMapping("Npc")
 public class NPCController 
 {
-    UUID ID;
+    UUID id;
     @Autowired
     NpcDataHandler dataHandler;
 

--- a/src/main/java/com/senecafoundation/NPCController.java
+++ b/src/main/java/com/senecafoundation/NPCController.java
@@ -20,7 +20,6 @@ import org.springframework.web.bind.annotation.RequestMethod;
 @RequestMapping("Npc")
 public class NPCController 
 {
-    UUID id;
     @Autowired
     NpcDataHandler dataHandler;
 
@@ -55,12 +54,13 @@ public class NPCController
         model.addAttribute("Npc", npc);
         return "create_Npc"; 
     }
-    @RequestMapping(value="/updateform", method = RequestMethod.PUT)
+    @RequestMapping(value="/updateform", method = RequestMethod.POST)
     public String change(NPC npc, BindingResult result, ModelMap model) {
         if (result.hasErrors()) {
-            return "Error";
+            return "error";
         }
         dataHandler.Update(npc);
+        model.addAttribute("Npc",npc);
         return "Npc";   
     }
 

--- a/src/main/java/com/senecafoundation/ResponseController.java
+++ b/src/main/java/com/senecafoundation/ResponseController.java
@@ -52,23 +52,15 @@ public class ResponseController
         model.addAttribute("response", response);
         return "create_response"; 
     }
-    @RequestMapping(value="/updateform", method = RequestMethod.PUT)
-    public String change(@ModelAttribute("response") Response response, BindingResult result, ModelMap model) {
+    @RequestMapping(value="/updateform", method = RequestMethod.POST)
+    public String change(Response response, BindingResult result, ModelMap model) {
         if (result.hasErrors()) {
-            return "Error";
+            return "error";
         }
         dataHandler.Update(response);
+        model.addAttribute("response",response);
         return "response";   
     }
-
-    // @RequestMapping(value ="/updateform", method = RequestMethod.PUT)
-    // public String change(@ModelAttribute("response") Response response, BindingResult result, ModelMap model) {
-    //     if (result.hasErrors()) {
-    //         return "error";
-    //     }
-    //     dataHandler.Update(response);
-    //     return "response"; 
-    // }
 
     @GetMapping("/deleteform")
     public String showFormDelete(Model model) {

--- a/src/main/java/com/senecafoundation/ScenarioController.java
+++ b/src/main/java/com/senecafoundation/ScenarioController.java
@@ -51,12 +51,13 @@ public class ScenarioController
         model.addAttribute("scenario", scenario);
         return "create_scenario"; 
     }
-    @RequestMapping(value="/updateform", method = RequestMethod.PUT)
-    public String change(@ModelAttribute("scenario") Scenario scenario, BindingResult result, ModelMap model) {
+    @RequestMapping(value="/updateform", method = RequestMethod.POST)
+    public String change(Scenario scenario, BindingResult result, ModelMap model) {
         if (result.hasErrors()) {
-            return "Error";
+            return "error";
         }
         dataHandler.Update(scenario);
+        model.addAttribute("scenario",scenario);
         return "scenario";   
     }
 

--- a/src/main/java/com/senecafoundation/Scene/Choice.java
+++ b/src/main/java/com/senecafoundation/Scene/Choice.java
@@ -43,7 +43,7 @@ public class Choice
 
     public void setId(UUID randomUUID) 
     {
-        
+        this.id = randomUUID;
     }
 
     public int getChoiceOptionNumber() {

--- a/src/main/resources/templates/create_Npc.html
+++ b/src/main/resources/templates/create_Npc.html
@@ -9,7 +9,7 @@
             
              <table> 
                  <tr>
-                    <td><input type="hidden" th:field="*{ID}" /></td>
+                    <td><input type="hidden" th:field="*{id}" /></td>
                  </tr> 
                 <tr>
                     <td><label>Name</label></td>

--- a/src/main/resources/templates/create_Npc.html
+++ b/src/main/resources/templates/create_Npc.html
@@ -5,12 +5,9 @@
 </head>
     <body>
         <h3>Welcome to the NPC Creation Screen</h3>
-        <form action="#" th:action="@{/Npc/createform}" method="post" th:object="${Npc}">
-            
-             <table> 
-                 <tr>
-                    <td><input type="hidden" th:field="*{id}" /></td>
-                 </tr> 
+        <form action="#" th:action="${Npc.id == null} ? @{/Npc/createform} : @{/Npc/updateform}" method="post" th:object="${Npc}">
+            <input type="hidden" th:field="*{id}" />
+            <table> 
                 <tr>
                     <td><label>Name</label></td>
                     <td><input type="text" th:field="*{name}" /></td>

--- a/src/main/resources/templates/create_choice.html
+++ b/src/main/resources/templates/create_choice.html
@@ -6,8 +6,9 @@
 </head>
     <body>
         <h3>Welcome, Create a Choice</h3>
-        <form action="#" th:action="@{/choice/createform}" method="post" th:object="${choice}">
-             <table> 
+        <form action="#" th:action="${choice.id == null} ? @{/choice/createform} : @{/choice/updateform}" method="post" th:object="${choice}">
+            <input type="hidden" th:field="*{id}" />
+            <table> 
                 <tr>
                     <td><label>ChoiceOptionNumber</label></td>
                     <td><input type="text" th:field="*{choiceOptionNumber}" /></td>

--- a/src/main/resources/templates/create_response.html
+++ b/src/main/resources/templates/create_response.html
@@ -5,8 +5,9 @@
 </head>
     <body>
         <h3>Welcome, Create a Response</h3>
-        <form action="#" th:action="@{/response/createform}" method="post" th:object="${response}">
-             <table> 
+        <form action="#" th:action="${response.id == null} ? @{/response/createform} : @{/response/updateform}" method="post" th:object="${response}">
+            <input type="hidden" th:field="*{id}" />
+            <table> 
                 <tr>
                     <td><label>ResponseText</label></td>
                     <td><input type="text" th:field="*{responseText}" /></td>

--- a/src/main/resources/templates/create_scenario.html
+++ b/src/main/resources/templates/create_scenario.html
@@ -5,8 +5,9 @@
 </head>
     <body>
         <h3>Welcome, Enter Scenario</h3>
-        <form action="#" th:action="@{/scenario/createform}" method="post" th:object="${scenario}">
-             <table> 
+        <form action="#" th:action="${scenario.id == null} ? @{/scenario/createform} : @{/scenario/updateform}" method="post" th:object="${scenario}">
+            <input type="hidden" th:field="*{id}" />
+            <table> 
                 <tr>
                     <td><label>Scene Name</label></td>
                     <td><input type="text" th:field="*{sceneName}" /></td>

